### PR TITLE
sort the list result for ps,sandboxes,images

### LINKED ISSUE #168: list result should be sorted.
Currently, `ps`, `sandboxes` and `images` result changes everytime.

### PR DESCRIPTION
**Pull request by yanxuean**
fix #168
Signed-off-by: yanxuean <yan.xuean@zte.com.cn>

### CODE REVIEW

**Comment by yanxuean** [comment]
test case
1. for container
```
root@ubuntu:/home/cloud/test# crictl ps
DEBU[0000] ListContainerRequest: &ListContainersRequest{Filter:&ContainerFilter{Id:,State:nil,PodSandboxId:,LabelSelector:map[string]string{},},} 
DEBU[0000] ListContainerResponse: &ListContainersResponse{Containers:[&Container{Id:11053ee6e8789351163f75720c364237c6f36e1c822525075a7546d7153600fd,PodSandboxId:dba535aa7bf2ea6675a4dce63457b11cf2c2fffdc151c1489f2478f4a5d50dfd,Metadata:&ContainerMetadata{Name:busybox2,Attempt:0,},Image:&ImageSpec{Image:busybox,},ImageRef:sha256:54511612f1c4d97e93430fc3d5dc2f05dfbe8fb7e6259b7351deeca95eaf2971,State:CONTAINER_CREATED,CreatedAt:1509518392514969405,Labels:map[string]string{},Annotations:map[string]string{},} &Container{Id:11c4699514bea2acdf83e0d063a9a5b0ffab50876591c4032ba022cedda393c4,PodSandboxId:dba535aa7bf2ea6675a4dce63457b11cf2c2fffdc151c1489f2478f4a5d50dfd,Metadata:&ContainerMetadata{Name:busybox1,Attempt:0,},Image:&ImageSpec{Image:busybox,},ImageRef:sha256:54511612f1c4d97e93430fc3d5dc2f05dfbe8fb7e6259b7351deeca95eaf2971,State:CONTAINER_CREATED,CreatedAt:1509518419261782382,Labels:map[string]string{},Annotations:map[string]string{},}],} 
CONTAINER ID                                                       CREATED             STATE               NAME
11c4699514bea2acdf83e0d063a9a5b0ffab50876591c4032ba022cedda393c4   11 seconds ago      CONTAINER_CREATED   busybox1
11053ee6e8789351163f75720c364237c6f36e1c822525075a7546d7153600fd   37 seconds ago      CONTAINER_CREATED   busybox2
root@ubuntu:/home/cloud/test# 
```
2. for image
```
root@ubuntu:/home/cloud/test# crictl images --digests
DEBU[0000] ListImagesRequest: &ListImagesRequest{Filter:&ImageFilter{Image:&ImageSpec{Image:,},},} 
DEBU[0000] ListImagesResponse: &ListImagesResponse{Images:[&Image{Id:sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27,RepoTags:[docker.io/library/busybox:1.26],RepoDigests:[docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642],Size_:701277,Uid:nil,Username:,} &Image{Id:sha256:54511612f1c4d97e93430fc3d5dc2f05dfbe8fb7e6259b7351deeca95eaf2971,RepoTags:[docker.io/library/busybox:latest docker.io/library/busybox:1.27.2],RepoDigests:[docker.io/library/busybox@sha256:3e8fa85ddfef1af9ca85a5cfb714148956984e02f00bec3f7f49d3925a91e0e7],Size_:719904,Uid:nil,Username:,} &Image{Id:sha256:320aedbfaad3d719b5e6a711834d2d06e2b0a172ce6fdadf84c640b6136914ed,RepoTags:[gcr.io/google_containers/pause:3.0],RepoDigests:[],Size_:312194,Uid:nil,Username:,} &Image{Id:sha256:350b164e7ae1dcddeffadd65c76226c9b6dc5553f5179153fb0e36b78f2a5e06,RepoTags:[gcr.io/google_containers/pause:latest],RepoDigests:[gcr.io/google_containers/pause@sha256:a78c2d6208eff9b672de43f880093100050983047b7b0afe0217d3656e1b0d5f],Size_:72335,Uid:nil,Username:,} &Image{Id:sha256:0300bfa969e985f2a2d35cc516ccaff1f132f3cc8362253926a8bdcb58c7d59f,RepoTags:[gcr.io/google_containers/pause:2.0],RepoDigests:[],Size_:341337,Uid:nil,Username:,}],} 
IMAGE                            TAG                 DIGEST                                                                    IMAGE ID            SIZE
docker.io/library/busybox        latest              sha256:3e8fa85ddfef1af9ca85a5cfb714148956984e02f00bec3f7f49d3925a91e0e7   54511612f1c4d       720kB
docker.io/library/busybox        1.27.2              sha256:3e8fa85ddfef1af9ca85a5cfb714148956984e02f00bec3f7f49d3925a91e0e7   54511612f1c4d       720kB
docker.io/library/busybox        1.26                sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642   c30178c5239f2       701kB
gcr.io/google_containers/pause   2.0                 <none>                                                                    0300bfa969e98       341kB
gcr.io/google_containers/pause   3.0                 <none>                                                                    320aedbfaad3d       312kB
gcr.io/google_containers/pause   latest              sha256:a78c2d6208eff9b672de43f880093100050983047b7b0afe0217d3656e1b0d5f   350b164e7ae1d       72.3kB
root@ubuntu:/home/cloud/test# 

```
3. for sandbox
```
root@ubuntu:/home/cloud/test# crictl sandboxes
DEBU[0000] ListPodSandboxRequest: &ListPodSandboxRequest{Filter:&PodSandboxFilter{Id:,State:nil,LabelSelector:map[string]string{},},} 
DEBU[0000] ListPodSandboxResponse: &ListPodSandboxResponse{Items:[&PodSandbox{Id:dba535aa7bf2ea6675a4dce63457b11cf2c2fffdc151c1489f2478f4a5d50dfd,Metadata:&PodSandboxMetadata{Name:busybox-sandbox,Uid:hdishd83djaidwnduwk28bcsb,Namespace:default,Attempt:1,},State:SANDBOX_READY,CreatedAt:1509509486978741638,Labels:map[string]string{},Annotations:map[string]string{},} &PodSandbox{Id:7595b8b07d43987fefa06a2d27fbce4ced376f0517a771f015e0ba9a36f15bb7,Metadata:&PodSandboxMetadata{Name:busybox-sandbox2,Uid:hdishd83djaidwnduwk28bcsc,Namespace:default,Attempt:1,},State:SANDBOX_READY,CreatedAt:1509524324987042544,Labels:map[string]string{},Annotations:map[string]string{},}],} 
SANDBOX ID                                                         NAME                STATE
dba535aa7bf2ea6675a4dce63457b11cf2c2fffdc151c1489f2478f4a5d50dfd   busybox-sandbox     SANDBOX_READY
7595b8b07d43987fefa06a2d27fbce4ced376f0517a771f015e0ba9a36f15bb7   busybox-sandbox2    SANDBOX_READY
root@ubuntu:/home/cloud/test# crictl sandboxes
DEBU[0000] ListPodSandboxRequest: &ListPodSandboxRequest{Filter:&PodSandboxFilter{Id:,State:nil,LabelSelector:map[string]string{},},} 
DEBU[0000] ListPodSandboxResponse: &ListPodSandboxResponse{Items:[&PodSandbox{Id:7595b8b07d43987fefa06a2d27fbce4ced376f0517a771f015e0ba9a36f15bb7,Metadata:&PodSandboxMetadata{Name:busybox-sandbox2,Uid:hdishd83djaidwnduwk28bcsc,Namespace:default,Attempt:1,},State:SANDBOX_READY,CreatedAt:1509524324987042544,Labels:map[string]string{},Annotations:map[string]string{},} &PodSandbox{Id:dba535aa7bf2ea6675a4dce63457b11cf2c2fffdc151c1489f2478f4a5d50dfd,Metadata:&PodSandboxMetadata{Name:busybox-sandbox,Uid:hdishd83djaidwnduwk28bcsb,Namespace:default,Attempt:1,},State:SANDBOX_READY,CreatedAt:1509509486978741638,Labels:map[string]string{},Annotations:map[string]string{},}],} 
SANDBOX ID                                                         NAME                STATE
dba535aa7bf2ea6675a4dce63457b11cf2c2fffdc151c1489f2478f4a5d50dfd   busybox-sandbox     SANDBOX_READY
7595b8b07d43987fefa06a2d27fbce4ced376f0517a771f015e0ba9a36f15bb7   busybox-sandbox2    SANDBOX_READY
root@ubuntu:/home/cloud/test#
```

File: cmd/crictl/container.go [hunk 1]
```diff
@@ -524,6 +536,7 @@ func ListContainers(client pb.RuntimeServiceClient, opts listOptions) error {
 	if err != nil {
 		return err
 	}
+	sort.Sort(containerByName(r.Containers))
```
Comment (feiskyer) [inline cmd/crictl/container.go]: Seems docker ps sorts output via createdAt , could you also sort the output this way?

Comment (yanxuean) [reply cmd/crictl/container.go]: Of course. done.

Comment (yanxuean) [reply cmd/crictl/container.go]: I have not studied the docker source code.

File: cmd/crictl/sandbox.go [hunk 2]
```diff
@@ -30,6 +31,20 @@ import (
 	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
 )
 
+type sandboxBySort []*pb.PodSandbox
+
+func (a sandboxBySort) Len() int      { return len(a) }
```
Comment (Random-Liu) [inline cmd/crictl/sandbox.go]: We should also sort sandbox by created time.

File: cmd/crictl/image.go [hunk 3]
```diff
@@ -35,6 +36,20 @@ const (
 	truncatedImageIDLen = 13
 )
 
+type imageBySort []*pb.Image
```
Comment (Random-Liu) [inline cmd/crictl/image.go]: We don't have created time for image. :( I think it's fine to sort by tag/digest for now.

Comment (Random-Liu) [reply cmd/crictl/image.go]: Why compare digest first? I think we care more about tag, right?

Comment (yanxuean) [reply cmd/crictl/image.go]: done.

File: cmd/crictl/sandbox.go (hunk 2 above)
Comment (yanxuean) [reply cmd/crictl/sandbox.go]: Only compare CreatAt?

File: cmd/crictl/container.go (hunk 1 above)
Comment (Random-Liu) [reply cmd/crictl/container.go]: The name should be `containerByCreated` now.

File: cmd/crictl/image.go [hunk 4]
```diff
@@ -105,6 +120,7 @@ var listImageCommand = cli.Command{
 		if err != nil {
 			return fmt.Errorf("listing images failed: %v", err)
 		}
+		sort.Sort(imageBySort(r.Images))
```
Comment (Random-Liu) [inline cmd/crictl/image.go]: `imageByRef`?

File: cmd/crictl/sandbox.go (hunk 2 above)
Comment (Random-Liu) [reply cmd/crictl/sandbox.go]: Make sense. I think you are right.

File: cmd/crictl/container.go (hunk 1 above)
Comment (yanxuean) [reply cmd/crictl/container.go]: done

File: cmd/crictl/image.go (hunk 4 above)
Comment (yanxuean) [reply cmd/crictl/image.go]: done

**Pull request merged**